src/golibmc.go

```diff
@@ -346,7 +353,7 @@ func (client *Client) newConn() (*conn, error) {
 	)
 
 	cn.configHashFunction(int(hashFunctionMapping[client.hashFunc]))
-	if client.retryTimeout > 0 {
+	if client.retryTimeout > 0 || client.notWaitForRetryTimeout {
```

Comment:
How does that make sense if the `retryTimeout` is still greater than 0?

Reply:
Oh.. I see. Why not just change `client.retryTimeout > 0` to `client.retryTimeout >= 0` ?

Reply:
If `client.ConfigTimeout (golibmc.RetryTimeout, ~)` is not called, `client.RetryTimeout` is 0, but in this case it was set to 5 which is default value until this PR was merged It was.
If I make a fix like `> =`, we have implemented another function, as it will cause a change to the user that used to do so.

Reply:
> If client.ConfigTimeout (golibmc.RetryTimeout, ~) is not called, client.RetryTimeout is 0, but in this case it was set to 5 which is default value until this PR was merged It was.

I don't understand why you said it is 0. If not called, it should be 5 (`#define MC_DEFAULT_RETRY_TIMEOUT 5`). After the PR is merged, it's still 5.



It looks like because of a bug introduced in https://github.com/douban/libmc/commit/3b595db54f4997e2957616a82cc9ef17d1b9f341#diff-b21c9a32b7563e26c49d7f215bc5a656L584 . After that commit, the `.ConfigTimeout` function is not working anymore if called after the connection is established since timeout values won't be passed into the C++ layer. Actually the `.ConfigTimeout` has never been used in the golang code anymore, so how did you guys override the timeout values?

I think changing this to`>=` is acceptable and compatible with previous releases as the default behaviour is not changed at all.

cc @ariesdevil @MOON-CLJ

Reply:
Calling `c.client_config` line is [here](https://github.com/douban/libmc/blob/1b185ad77bf285f37d944696fe2045b36a6bbace/src/golibmc.go#L356-L364).
And setting `MC_DEFAULT_RETRY_TIMEOUT` is [here](https://github.com/douban/libmc/blob/1b185ad77bf285f37d944696fe2045b36a6bbace/src/Connection.cpp#L26).

Reply:
When implementing a connection pool, I fixed `ConfigTimeout` so that `C. client_config` will be called at the time when a new connection is created.

Reply:
@jumpeiMano how about init client.RetryTimeout to MC_DEFAULT_RETRY_TIMEOUT? and "just change client.retryTimeout > 0 to client.retryTimeout >= 0"。I also think notWaitForRetryTimeout is not necessary。